prboom2/src/gl_main.c

```diff
@@ -2328,20 +2328,29 @@ static void gld_DrawSprite(GLSprite *sprite)
 
 static void gld_AddHealthBar(mobj_t* thing, GLSprite *sprite)
 {
-  if (((thing->flags & (MF_COUNTKILL | MF_CORPSE)) == MF_COUNTKILL) && (thing->health > 0))
+  if ((thing->flags & MF_SHOOTABLE) && thing->info->spawnhealth > 0 &&
```

Comment:
Still not convinced of this change, though. I wouldn't want health bars for barrels.

Reply:
Knowing the barrel's state is just as essential as knowing a monster's health state; a barrel can usually take a lowpower hit, sometimes two. This influences whether it's "reasonably safe" to speedrun past it. Or, strategy: tickling the barrel raises the chance it will definitely go off on the next hit, and if that final blow is dealt by a monster, that's free infighting for the speedrunner.

Reply:
I wouldn't expect to see health bars on barrels either.

Reply:
added a config option for this, `health_bar_shootables`.